bin/jsondiff.py

```diff
@@ -0,0 +1,113 @@
+import os
+import sys
+import argparse
+import diff
+import deepdiff
+
+def add_arguments():
+    """Returns the arguments provided by the user and parsed by the Argument Parser"""
+    parser = argparse.ArgumentParser(description='Provide two json files (.json) to check for differences', add_help=False)
+    options = parser.add_argument_group(title='Options')
+    options.add_argument('json_files',
+                         metavar='<json_file, json_file>',
+                         help='Specify two json files',
+                         type=validate_json_extension,
+                         nargs=2)
+    # Note: nargs='?' and const=value is for 0 or 1 arguments supplied. If 0 arguments provided, the const value will be used
+    options.add_argument('--rel_tol',
+                         metavar='<relative_error>',
+                         dest='relative_error',
+                         help='Specify the maximum relative error (default 1e-8)',
+                         type=float,
+                         nargs='?', 
+                         const=1e-8)
+    options.add_argument('--abs_tol',
+                         metavar='<absolute_error>',
+                         dest='absolute_error',
+                         help='Specify the maximum absolute error (default 1e-10)',
+                         type=float,
+                         nargs='?', 
+                         const=1e-10)
+    options.add_argument('-h', '--help', action='help', help='Displays CLI usage statement')
+    args, unknown = parser.parse_known_args()
+    
+    # Set arg parameter to None if no flag was supplied
+    if '--rel_tol' not in sys.argv[1:]:
+        args.relative_error = None
+    if '--abs_tol' not in sys.argv[1:]:
+        args.absolute_error = None
+    return args
+
+def validate_json_extension(json_file):
+    """Validates that the json file has an .json extension and path exists
+        json_file: the file to validate"""
+    return_code = 0
+    base, extension = os.path.splitext(json_file)
+    if extension.lower() not in ('.json'):
+        raise argparse.ArgumentTypeError('Invalid extension with "' + str(json_file) + '": Provide a file with a .json extension')
+    return json_file
+
+def validate_json_files_exist(args):
+    """Check if json files exist in the system path
+        args: arguments provided by the user"""
+    return_code = 0
+    are_files = list()
+    for json_file in args.json_files:
+        if not os.path.exists(json_file):
+            print('Path does not exist: ' + str(json_file))
+            return_code += 1
+    return return_code
```

Comment:
I like these functions and they will be more useful that just in this tool. So, can you add them to the moosetools directory and put them in a file called validate: i.e., moosetools/mooseutils/validate.py

Change "validate_json_extension" to "validate_extension" and require the extension to be provided as an argument, so we can use it for other file types.

Change "validate_json_files_exist" to "validate_files_exist".

Both should return an exit code 0 good, 1 bad. And both should have an additional flags: log_on_error and raise_on_error, obviously those should do what the names suggest.

```
validate_files_exist(*args, raise_on_error=False, log_on_error=True):
```

Reply:
Also, add some tests for these functions.